### DISCUSSION TITLE
How to get the Rect from text only, not Rect from "form text field"

### DISCUSSION THREAD

**Discussion by hoaile-novobi** [pymupdf/PyMuPDF#1892] [Looking for help] [answered]
Hi admin, I would like to get the Rect of text only in the Text Field Widget. Is there any method to do that ?

Here is my example code:

```
for field in fields:
  if field.field_name == key and content:
      field.field_value = "Jhon D Sample"
      field.update()
      name_block_rect = field.rect
      name_block_text_rect = ??? (Only rect for Jhon D Sample)
```

![image](https://user-images.githubusercontent.com/104294815/185898668-d2313712-b8cb-46dd-bb9e-2d37df25a209.png)

**Comment by JorjMcKie** [comment]
There is no support within the widget API.
But - at least in your case - font Helvetica is predictably being used. So you can compute the text length via get_text_length(text) plus providing the fontsize of the widget.

**Reply by hoaile-novobi** [reply]
How can i call this method ? Is it call via Text Widget ?

**Comment by JorjMcKie** [comment]
Transferring this to the Discussions tab.

**Comment by JorjMcKie** [comment, marked as answer]
This is the function with its defaults: `fitz.get_text_length(text, fontname='helv', fontsize=11, encoding=TEXT_ENCODING_LATIN)`.

**Reply by hoaile-novobi** [reply]
Can i use the font that embeded in pdf file?

```
cat_xref = template_pdf.pdf_catalog()
form_fonts = template_pdf.xref_get_key(cat_xref, "AcroForm/DR/Font")
form_fonts = form_fonts[1].split("/") if form_fonts[1] else []
```

I reference this code form the issue 1184, that is the font that name_block field use. Use this font will calculate more exactly

**Reply by JorjMcKie** [reply]
The function only works correctly for the Base-14 fonts (Times-Roman, Helvetica, Courier). If other fonts are used you must extract their binaries from the PDF and then define a `Font` object, which also has a text length calculator:
```python
font = fitz.Font(fontbuffer=<extracted font binary>)
font.text_length(text, fontsize=11)
```

**Reply by hoaile-novobi** [reply]
I call the `page.get_fonts()` func, the list is empty. But the code above is return a list of font
Here is the `form_fonts` return from the code i mentioned above: 

`['<<', 'Helv 10 0 R', 'HelveticaNeue 11 0 R', 'HelveticaNeue,Bold 75 0 R', 'HelveticaNeue,Italic 12 0 R', 'HelveticaNeue-Light 78 0 R', 'HelveticaNeue-ThinItalic 13 0 R', 'HelveticaNeueLTStd-Lt 14 0 R', 'HelveticaNeueLTStd-LtIt 87 0 R', 'HelveticaNeueLTStd-Roman 15 0 R', 'HighTowerText,Italic 16 0 R', 'ZaDb 17 0 R>>']`

I know exactly what the font name is, that is "HelveticaNeue,Bold'. How can i get the buffer of this font ?

**Reply by JorjMcKie** [reply]
In the list you find "75 0 R". Use `fff=doc.extract_font(75)`. The fontbuffer is ff[-1].

**Reply by hoaile-novobi** [reply]
Thank you so much
